core/src/test/java/org/bitcoinj/core/LegacyAddressTest.java

```diff
@@ -188,6 +188,21 @@ public void p2shAddress() {
         assertEquals("35b9vsyH1KoFT5a5KtrKusaCcPLkiSo1tU", c.toString());
     }
 
+    @Test
+    public void p2shAddressFromPrivateKey() {
+        final NetworkParameters networkParameters = MainNetParams.get();
+        final String privateKeyWIF = "L3E6dAmEwcRNbeUnzjd6YuhMm9fDxgzVmJHXAV1gtXm5e49Dctiv";
+
+        final ECKey key = DumpedPrivateKey.fromBase58(networkParameters, privateKeyWIF).getKey();
+        final Script script = ScriptBuilder.createP2SHOutputScript(key);
+
+        final LegacyAddress addressFromScriptHash = LegacyAddress.fromScriptHash(networkParameters, ScriptPattern.extractHashFromP2SH(script));
+        final LegacyAddress addressFromKey = LegacyAddress.fromKey(networkParameters, key, ScriptType.P2SH);
```

Comment:
Minor point, but `final` is probably overkill here. Everything is one test/method and it's clear that no variable is or will be set twice. (I think this also varies from the other code in bitcoinj)

Reply:
`final` is just a hit for JVM that this variable is final and it will not be reassigned, so it is not an overkill, but good practice.
Also, this provides better readability for developers (you do not need to search in code if a variable has been reassigned).